### PR TITLE
validation support tabbed conceptual

### PR DESCRIPTION
**Pull request by xichan-ms**
[AB#338036](https://dev.azure.com/ceapex/d3d54af3-265a-4f18-95f6-9a46397ca583/_workitems/edit/338036)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6797)

### CODE REVIEW

File: src/docfx/lib/markdown/MarkdigUtility.cs [hunk 1]
```diff
@@ -64,6 +64,25 @@ public static MonikerList GetZoneLevelMonikers(this MarkdownObject obj)
...
+                    var group = (TabGroupBlock)content.Parent;
+                    foreach (var item in group.Items)
+                    {
+                        if (item.Content == content)
+                        {
+                            return string.IsNullOrEmpty(item.Condition) ? item.Id : item.Id + "/" + item.Condition;
+                        }
+                    }
+                }
+            }
```
Comment (bianyifan) [inline src/docfx/lib/markdown/MarkdigUtility.cs:82]: Is it possible to add a backward reference from `TabContentBlock` to `TabItemBlock`, so we don't have to iterate through them?

Comment (xichan-ms) [reply src/docfx/lib/markdown/MarkdigUtility.cs:82]: Great idea, worth researching the possibility.

**Comment by xichan-ms** [comment]
Diff explanation:
- The errors are all deleting error, and no new appearing error(some in repo azure-devops-docs-pr are due to position changing).
- Checking the deleting error, they are by design and these errors are false positive due to ignore tabbed conceptual at the beginning.

**Review by bianyifan — COMMENTED** [review]
Looks good to me, @OsmondJiang @yufeih do you have any suggestion on this PR?

File: src/docfx/lib/markdown/MarkdigUtility.cs [hunk 2]
```diff
@@ -64,6 +64,19 @@ public static MonikerList GetZoneLevelMonikers(this MarkdownObject obj)
             return default;
         }
 
+        public static string? GetTabbedConceptual(this MarkdownObject obj)
+        {
+            foreach (var parent in obj.GetPathToRootInclusive())
```
Comment (fkpwolf) [inline src/docfx/lib/markdown/MarkdigUtility.cs:69]: How about use `FirstOrDefault` to find

File: src/docfx/lib/markdown/MarkdigUtility.cs [hunk 3]
```diff
@@ -64,6 +64,19 @@ public static MonikerList GetZoneLevelMonikers(this MarkdownObject obj)
...
+            {
+                if (parent is TabContentBlock content)
+                {
+                    var inLine = (LinkInline)content.TabTitle.Inline;
+                    return inLine.Url.Substring("#tab/".Length);
```
Comment (yufeih) [inline src/docfx/lib/markdown/MarkdigUtility.cs:74]: This method seems to be retrieving the `Id` property parsed out by `TabGroupAggregator.ParseHeading` method, so:
- Change the method name to `GetTabId`
- Add `Id` property to `TabContentBlock` and set it to the result of `TabGroupAggregator.ParseHeading`.

Comment (xichan-ms) [reply src/docfx/lib/markdown/MarkdigUtility.cs:74]: OK

File: src/docfx/lib/markdown/MarkdigUtility.cs (hunk 2 above)
Comment (xichan-ms) [reply src/docfx/lib/markdown/MarkdigUtility.cs:69]: OK

**Comment by xichan-ms** [comment]
Diff explanation:
- The errors are all deleting error, and no new appearing error(some in repo azure-devops-docs-pr are due to position changing).
- Checking the deleting error, they are by design and these errors are false positive due to ignore tabbed conceptual at the beginning.

**Review by yufeih — APPROVED** [review]

**Pull request merged**